### PR TITLE
Add documentation for `ruff-action` (GitHub Action!)

### PR DESCRIPTION
**Pull request by brucearctor**
Following up from now closed PR:  https://github.com/charliermarsh/ruff/pull/3820

This updates ruff docs and outlines the [Ruff-Action](https://github.com/chartboost/ruff-action).

This follows Pyright style:  https://github.com/microsoft/pyright/blob/main/docs/ci-integration.md

Ex ->
Pyright found at: https://github.com/microsoft/pyright
Pyright action found in separate 'org' at:  https://github.com/jakebailey/pyright-action

I'm glad we decided on this direction -- having setup this way is great -- I can more easily maintain/extend [ and as training for some of my engineers ], as well as is the first real step towards an open-source program which I've been meaning to get started with!  

Near-term follow up steps will be to increase documentation.  Do advise re: anything else that makes sense.  

Separately ->
Also, I'm not sure where you're based or your travel inclinations these days.  I'll be in Miami end-of-May, and am to give a conference talk to [Google Developer Student Club Leads](https://gdsc.community.dev/), [GDG organizers](https://gdg.community.dev/), [WTM Ambassadors](https://developers.google.com/womentechmakers/ambassadors), and [Google Developer Experts](https://goo.gle/gde).  [ I'm a GDE ].  The plan was on Python, CI, and in-this-case, likely `ruff`!  So, if you could potentially be around, and open to join, let's chat.

### CODE REVIEW

**Comment by ssweber** [comment]
thanks. giving it a whirl.

File: docs/usage.md [hunk 1]
```diff
@@ -43,3 +43,46 @@ reformatting.
...
+- version: Must be release available on PyPI. default, latest release of ruff. You can pin a version, or use any valid version specifier.
+- options: default,`check`
+- src: default, '.'
+
+```yaml
+- uses: chartboost/ruff-action@v1
+  with:
+    src: "./src" 
+    version: 0.0.259
+    options: --select B
```
[the remark this answers is not in the document]
Comment (brucearctor) [reply docs/usage.md]: 🤦 -- had updated the code, but not the docs with that change.  Docs now fixed, good catch.  Updated both here and in the action repo.

**Comment by charliermarsh** [comment]
What are your thoughts on ownership of the action? Would you be open to either or both of (1) PRing on https://github.com/charliermarsh/ruff-action, or (2) moving the action under our GitHub Org once it's setup in the next month or so? I'd like all the "official" Ruff projects to live under the same ownership structure. For example, right now, the pre-commit action is under https://github.com/charliermarsh/ruff-pre-commit and will move under the GitHub Org once it's setup.

**Comment by charliermarsh** [comment]
> Also, I'm not sure where you're based or your travel inclinations these days. I'll be in Miami end-of-May, and am to give a conference talk to [Google Developer Student Club Leads](https://gdsc.community.dev/), [GDG organizers](https://gdg.community.dev/), [WTM Ambassadors](https://developers.google.com/womentechmakers/ambassadors), and [Google Developer Experts](https://goo.gle/gde). [ I'm a GDE ]. The plan was on Python, CI, and in-this-case, likely ruff! So, if you could potentially be around, and open to join, let's chat.

That's amazing! It's unlikely I'll be able to join, I'm traveling for PyCon in late April and I have a baby at home so I'm trying to avoid traveling in back-to-back months. I'd love to help support however I can though.

**Comment by brucearctor** [comment]
> What are your thoughts on ownership of the action? Would you be open to either or both of (1) PRing on https://github.com/charliermarsh/ruff-action, or (2) moving the action under our GitHub Org once it's setup in the next month or so? I'd like all the "official" Ruff projects to live under the same ownership structure. For example, right now, the pre-commit action is under https://github.com/charliermarsh/ruff-pre-commit and will move under the GitHub Org once it's setup.

GitHub Org:  Setting one up is pretty straightforward, I mean if just about an org that isn't your name, that could be done today.  I imagine more you'll want to think about structure [ and legal aspects ], ex: perhaps you are going to try to build a business around ruff [ or 'high performance' python tooling ], or donate such things to a foundation and have it live there [ ex: Github/apache/ruff, github.com/apache/ruff-pre-commit, github.com/apache/ruff-action ]....  Happy to walk you/ruff through the ASF Incubating process, ex: https://incubator.apache.org/ if that's of interest.  

Since I can use this currently as a way to bootstrap an OSS program at Chartboost, I do like the thought of it living there initially, gives nice company talking points [ I recently started there, so a decent 'win' for me, and serves as foundation for doing more of similar ].  Donating to a proper organization also sounds sensible once the to-be-created org is ready... Again, that's a great step corporation-wise, not only openly sharing but also giving, esp. to a place that then would be a suitable long-term home :-)

**Comment by brucearctor** [comment]
I now already see 2 company blogposts!  1st being that chartboost created this usable thing to serve engineers in our company and the wider ruff community, and 2 that we donated to a good home/community for better/common support/utilization as sensible.

**Comment by charliermarsh** [comment]
Ok, that sounds reasonable! I do plan to setup the Org and move the official Ruff projects under it, it's just blocked on some other stuff and I want to do it "all at once", ideally before the end of April. I may ask if you'd be willing to transfer the project at that time, but for now we can link to the chartboost repo.

I'm gonna make some tweaks to the docs to ensure consistent voice and tone with the rest of the Ruff docs, you're welcome to pull in those changes or not to the chartboost repo, totally up to you!

**Comment by brucearctor** [comment]
Definitely...  can pull in improved docs.

**Pull request merged**